### PR TITLE
simplify distributed async code ( slightly ) 

### PR DESCRIPTION
**Pull request by vthacker**
use `CompletableFutures.successfulAsList`

### CODE REVIEW

File: kaldb/src/main/java/com/slack/kaldb/logstore/search/KaldbDistributedQueryService.java [hunk 1]
```diff
@@ -26,7 +26,7 @@
   // TODO: ChunkManager#QUERY_TIMEOUT_SECONDS and this could be unified?
   public static int READ_TIMEOUT_MS = 15000;
 
-  public static final KaldbSearch.SearchResult error =
+  public static final KaldbSearch.SearchResult emptyResult =
```
Comment (vthacker) [inline kaldb/src/main/java/com/slack/kaldb/logstore/search/KaldbDistributedQueryService.java:29]: rename

File: kaldb/src/main/java/com/slack/kaldb/logstore/search/KaldbDistributedQueryService.java [hunk 2]
```diff
@@ -35,7 +35,8 @@
   private CompletableFuture<List<KaldbSearch.SearchResult>> distributedSearch(
       KaldbSearch.SearchRequest request) {
 
-    List<CompletableFuture<KaldbSearch.SearchResult>> futures = new ArrayList<>(servers.size());
+    List<CompletableFuture<KaldbSearch.SearchResult>> queryServers =
```
Comment (vthacker) [inline kaldb/src/main/java/com/slack/kaldb/logstore/search/KaldbDistributedQueryService.java:38]: rename

File: kaldb/src/main/java/com/slack/kaldb/logstore/search/KaldbDistributedQueryService.java [hunk 3]
```diff
@@ -46,18 +47,14 @@
...
-        .stream()
-        .map(
-            result ->
-                result.exceptionally(
-                    ex -> {
-                      LOG.error("Node failed to respond ", ex);
-                      return error;
-                    }))
-        .collect(CompletableFutures.joinList());
+    return CompletableFutures.successfulAsList(
```
Comment (vthacker) [inline kaldb/src/main/java/com/slack/kaldb/logstore/search/KaldbDistributedQueryService.java:52]: simplification

File: kaldb/src/main/java/com/slack/kaldb/logstore/search/KaldbDistributedQueryService.java [hunk 4]
```diff
@@ -1,11 +1,11 @@
 package com.slack.kaldb.logstore.search;
 
 import com.linecorp.armeria.client.Clients;
-import com.linecorp.armeria.internal.shaded.futures.CompletableFutures;
 import com.slack.kaldb.logstore.LogMessage;
 import com.slack.kaldb.proto.service.KaldbSearch;
 import com.slack.kaldb.proto.service.KaldbServiceGrpc;
 import com.slack.kaldb.server.KaldbQueryServiceBase;
+import com.spotify.futures.CompletableFutures;
```
Comment (vthacker) [inline kaldb/src/main/java/com/slack/kaldb/logstore/search/KaldbDistributedQueryService.java:8]: use the correct import

**Review by bryanlb — APPROVED** [review]

**Review by mansu — APPROVED** [review]

**Pull request merged**